### PR TITLE
[sw/device] Trivial fixes courtesy of clang-tidy

### PR DESCRIPTION
**Pull request by dmcardle**
I temporarily enabled strict prototypes and pedantic warnings before running clang tidy:

```bash
./bazelisk.sh query 'kind(cc_, //sw/...)' | xargs ./bazelisk.sh build -k --config=clang_tidy_fix //sw/device/...
./bazelisk.sh query 'kind(cc_, //sw/...)' | xargs ./bazelisk.sh build -k --config=clang_tidy_fix --config=riscv32 //sw/device/...
```

This is how I modified the compiler flags:

```diff
diff --git a/.bazelrc b/.bazelrc
index 497d224b5..1771f93d5 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -11,10 +11,29 @@ build --conlyopt='-std=gnu11'
 # disassemblies when compilation mode is fastbuild.
 build --strip='never'
 
-# Bazel embedded enables the following feature which along with -std=c11 and
-# our codebase generates a lot of warnings by setting the -Wpedantic flag
-build --features=-all_warnings
-build --features=-pedantic_warnings
+# Override default enablement of flags from @crt//common to control C compiler
+# warnings.
+#
+# TODO(#12553) Remove these explicit feature flags once the features in @crt//
+# are enabled by default.
+build --features=pedantic_warnings
+build --features=strict_prototypes_warnings
+build --features=covered_switch_default_warnings
+build --features=extra_warnings
+build --features=clang_covered_switch_default_warnings
+build --features=implicit_conversion_warnings
+build --features=implicit_fallthrough_warnings
+build --features=invalid_pch_warnings
+build --features=switch_default_warnings
+build --features=no_missing_field_initializers_warning
+build --features=no_sign_compare_warning
+build --features=no_unused_function_warning
+build --features=no_unused_parameter_warning
+# Override default enablement of flags from @crt//embedded to control C compiler
+# warnings.
+build --features=clang_gnu_warnings
+build --features=no_gnu_zero_variadic_macro_arguments_warning
+build --features=no_gnu_statement_expression_from_macro_expansion
```

### CODE REVIEW

File: sw/device/sca/ecc256_sign_serial.c [hunk 1]
```diff
@@ -96,7 +96,7 @@ uint32_t ecc256_msg[kEcc256NumWords] = {
 };
 
 // p256_ecdsa_sca has randomnization removed.
-OTBN_DECLARE_APP_SYMBOLS(p256_ecdsa_sca);
+OTBN_DECLARE_APP_SYMBOLS(p256_ecdsa_sca)
```
Comment (alphan) [inline sw/device/sca/ecc256_sign_serial.c:99]: can we update the macro, e.g. remove the last semicolon, for the sake of consistency?

File: sw/device/sca/ecc384_serial.c [hunk 2]
```diff
@@ -78,7 +78,7 @@ uint32_t ecc384_msg[12] = {
 };
 
 // p384_ecdsa_sca has randomnization removed.
-OTBN_DECLARE_APP_SYMBOLS(p384_ecdsa_sca);
+OTBN_DECLARE_APP_SYMBOLS(p384_ecdsa_sca)
```
Comment (alphan) [inline sw/device/sca/ecc384_serial.c:81]: same here

**Review by alphan — APPROVED** [review]
Thanks @dmcardle !

**Comment by dmcardle** [comment]
Oh neat, fixing the strict prototype issue on `static void ecc256_ecdsa(void)` created a compiler error. Without the `void` parameter list, this error doesn't appear:

```
ERROR: /home/dan/REPOS/opentitan/sw/device/sca/BUILD:87:23: Compiling sw/device/sca/ecc256_sign_serial.c failed: (Exit 1): clang failed: error executing command external/crt/toolchains/lowrisc_rv32imcb/wrappers/clang -MD -MF bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/sca/_objs/ecc256_sign_serial_sim_verilator.elf/ecc256_sign_serial.d ... (remaining 70 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
sw/device/sca/ecc256_sign_serial.c:281:48: error: incompatible function pointer types passing 'void (void)' to parameter of type 'simple_serial_command_handler' (aka 'void (*)(const unsigned char *, unsigned int)') [-Wincompatible-function-pointer-types]
  SS_CHECK(simple_serial_register_handler('p', ecc256_ecdsa) ==
                                               ^~~~~~~~~~~~
./sw/device/sca/lib/simple_serial.h:31:11: note: expanded from macro 'SS_CHECK'
    if (!(condition)) {                              \
          ^~~~~~~~~
./sw/device/sca/lib/simple_serial.h:94:48: note: passing argument to parameter 'handler' here
    uint8_t cmd, simple_serial_command_handler handler);
                                               ^
```

Fixing this by adding the missing parameters.

**Pull request merged**